editor/editor_properties_array_dict.cpp

```diff
@@ -517,6 +543,33 @@ void EditorPropertyArray::drop_data_fw(const Point2 &p_point, const Variant &p_d
 		emit_changed(get_edited_property(), array, "", false);
 		update_property();
 	}
+
+	if (drag_data.has("type") && String(drag_data["type"]) == "nodes") {
+		Array nodes = drag_data["nodes"];
+		Array nodes_paths = Array();
+		EditorPropertyNodePath prop = EditorPropertyNodePath();
```

Comment:
You're constructing a value of a complex type here, and then getting a base object, where is that node coming from? Was this intended to be a cast of some kind?

Reply:
Initially I duplicated the `get_base_node()` function from `EditorPropertyNodePath` inside of the `EditorPropertyArray`  class so it had its own. I did it this way because I could not access the one from `EditorPropertyNodePath` and so that's how I solved it initally; but I was wondering if there was a way to avoid essentialy duplicating that function, so instead I made the other one public and did it this way and with basic testing it was working the same way but with a bit less code so I thought "why not", but maybe it is bad... Do you prefer that `EditorPropertyArray` have it's own `get_base_node()` as I did it initially ?

Reply:
You can't do it like this, you'll always get some null node, you're just default constructing the type, and doing so in an unsafe way, you should always construct things as `memnew`, but in this case that's irrelevant, the solution here is to write your own method to do this, the `EditorPropertyNodePath` expects to have a node associated with it

Reply:
Ok I will revert to how I did it initially then plus the change you suggested 👍